OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/BaseBuilderQueueManager.cs

```diff
@@ -218,14 +229,28 @@ ActorInfo GetProducibleBuilding(HashSet<string> actors, IEnumerable<ActorInfo> b
 
 		bool HasSufficientPowerForActor(ActorInfo actorInfo)
 		{
-			return playerPower == null || (actorInfo.TraitInfos<PowerInfo>().Where(i => i.EnabledByDefault)
-				.Sum(p => p.Amount) + playerPower.ExcessPower) >= baseBuilder.Info.MinimumExcessPower;
+			return playerPower == null || !actorInfo.HasTraitInfo<PowerInfo>() ||
```

Comment:
Isn't adding `HasTraitInfo` here just adding one more trait lookup? (The `.Where(...).Sum(...)` should result in 0 if there is no trait, which should be fine already?)

Reply:
The idea was to not have the bot require excess power for brick walls. I removed it now as it was an incomplete workaround for power plants obstructing the walls in some cases and I didn't want to wait during testing. I think in skirmish this delay of the wall building shouldn't give the AI a noticeable disadvantage.